### PR TITLE
Patch that add support for avocado to run default as legacy runner

### PR DESCRIPTION
**Pull request by PraveenPenguin**
This patch address to have defaulted to legacy runner as test repo
tests are written based on serial run

TODO#
add suport for nrunner

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
@narasimhan-v can you please have look at this PR as it has support to unblock misc test run using the latest avocado , nrunner we can explore later  as adding actual support

File: avocado-setup.py [hunk 1]
```diff
@@ -335,12 +338,16 @@ def bootstrap(enable_kvm=False, guest_os=None):
...
 
 
-def run_test(testsuite, avocado_bin):
+def run_test(testsuite, avocado_bin, nrunner):
     """
     To run given testsuite
     :param testsuite: Testsuite object which has details about the tests
     :param avocado_bin: Executable path of avocado
     """
+    nrunner = ''
```
Comment (narasimhan-v) [inline avocado-setup.py:347]: Can we do this via default param in function definition ?

Comment (narasimhan-v) [reply avocado-setup.py:347]: Also, please handle this as a binary as per the definition in argparse

**Review by narasimhan-v — CHANGES_REQUESTED** [review]

**Comment by PraveenPenguin** [comment]
@narasimhan-v adopted the review comment please have a look

**Comment by PraveenPenguin** [comment]
```
root@system  tests]# python avocado-setup.py  --run-suite host_mine
13:24:36 INFO    : Check for environment
13:24:37 INFO    : Creating temporary mux dir
13:24:39 INFO    : 
13:24:39 INFO    : Running Host Tests Suite mine_mine
13:24:39 INFO    : Running: /usr/local/bin/avocado run --test-runner runner  avocado-misc-tests/generic/connectathon.py avocado-misc-tests/generic/interbench.py avocado-misc-tests/generic/openblas.py avocado-misc-tests/kernel/tlbflush.py --force-job-id 25159e77e2ab0f6dfead45f244c5bbdc51910515                  --job-results-dir /root/tests/results  
Fetching asset from /root/tests/tests/avocado-misc-tests/generic/interbench.py:Interbench.test
Fetching asset from /root/tests/tests/avocado-misc-tests/generic/openblas.py:Openblas.test
JOB ID     : 25159e77e2ab0f6dfead45f244c5bbdc51910515
JOB LOG    : /root/tests/results/job-2021-11-08T13.24-25159e7/job.log
 (1/4) avocado-misc-tests/generic/connectathon.py:Connectathon.test: PASS (3.55 s)
 (2/4) avocado-misc-tests/generic/interbench.py:Interbench.test:
 ```

**Review by narasimhan-v — APPROVED** [review]
@PraveenPenguin please attach run log

**Comment by PraveenPenguin** [comment]
```
[root@system tests]# python avocado-setup.py --run-suite host_mine
13:43:13 INFO    : Check for environment
13:43:13 INFO    : Creating temporary mux dir
13:43:15 INFO    : 
13:43:15 INFO    : Running Host Tests Suite mine_mine
13:43:15 INFO    : Running: /usr/local/bin/avocado run --test-runner runner  avocado-misc-tests/cpu/ppc64_cpu_test.py avocado-misc-tests/ras/ras.py avocado-misc-tests/perf/perfmon.py --force-job-id 3d1acf9c2495007b4d2e757f5fa244f3f5031f84                  --job-results-dir /root/tests/results  
JOB ID     : 3d1acf9c2495007b4d2e757f5fa244f3f5031f84
JOB LOG    : /root/tests/results/job-2021-11-08T13.43-3d1acf9/job.log
 (01/15) avocado-misc-tests/cpu/ppc64_cpu_test.py:PPC64Test.test_cmd_options: PASS (6.27 s)
 (02/15) avocado-misc-tests/cpu/ppc64_cpu_test.py:PPC64Test.test_smt_loop: PASS (250.32 s)
 (03/15) avocado-misc-tests/ras/ras.py:RASTools.test1_set_poweron_time: PASS (1.27 s)
 (04/15) avocado-misc-tests/ras/ras.py:RASTools.test2_sys_ident_tool: PASS (1.09 s)
 (05/15) avocado-misc-tests/ras/ras.py:RASTools.test3_lsmcode: PASS (1.66 s)
 (06/15) avocado-misc-tests/ras/ras.py:RASTools.test4_drmgr: PASS (1.16 s)
 (07/15) avocado-misc-tests/ras/ras.py:RASTools.test5_lsprop: PASS (1.15 s)
 (08/15) avocado-misc-tests/ras/ras.py:RASTools.test6_lsslot: PASS (1.37 s)
 (09/15) avocado-misc-tests/ras/ras.py:RASTools.test7_lsvio: PASS (1.15 s)
 (10/15) avocado-misc-tests/ras/ras.py:RASTools.test8_nvram: PASS (1.15 s)
 (11/15) avocado-misc-tests/ras/ras.py:RASTools.test9_ofpathname: PASS (1.43 s)
 (12/15) avocado-misc-tests/ras/ras.py:RASTools.test11_rtas_ibm_get_vpd: PASS (1.09 s)
 (13/15) avocado-misc-tests/ras/ras.py:RASTools.test12_rtas_errd_and_rtas_dump: PASS (2.01 s)
 (14/15) avocado-misc-tests/ras/ras.py:RASTools.test13_rtas_event_decode: PASS (1.07 s)
 (15/15) avocado-misc-tests/perf/perfmon.py:Perfmon.test: PASS (14.07 s)
RESULTS    : PASS 15 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/tests/results/job-2021-11-08T13.43-3d1acf9/results.html
JOB TIME   : 295.02 s
13:48:16 INFO    : 
13:48:16 INFO    : Summary of test results can be found below:
TestSuite                                                                                 TestRun    Summary
 
host_mine_mine                                                                            Run        Successfully executed
/root/tests/results/job-2021-11-08T13.43-3d1acf9/job.log
| PASS 15 || CANCEL 0 || ERRORS 0 || FAILURES 0 || SKIP 0 || WARN 0 || INTERRUPT 0 |
13:48:16 INFO    : Removing temporary mux dir
[root@system tests]# 
````

**Comment by PraveenPenguin** [comment]
```
[root@system tests]# cat avocado-wrapper.log 
2021-11-08 14:58:33 DEBUG   : [helper.py:49] Running pip3 --help >/dev/null 2>&1||(curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && python3 ./get-pip.py)
2021-11-08 14:58:34 DEBUG   : [helper.py:55] 
2021-11-08 14:58:34 INFO    : [avocado-setup.py:125] Check for environment
2021-11-08 14:58:34 DEBUG   : [helper.py:49] Running rpm -q gcc
2021-11-08 14:58:35 DEBUG   : [helper.py:55] gcc-8.5.0-3.el8.ppc64le
2021-11-08 14:58:35 DEBUG   : [helper.py:49] Running rpm -q python36-devel
2021-11-08 14:58:35 DEBUG   : [helper.py:55] python36-devel-3.6.8-37.module+el8.5.0+10916+41bd434d.ppc64le
2021-11-08 14:58:35 DEBUG   : [helper.py:49] Running rpm -q xz-devel
2021-11-08 14:58:35 DEBUG   : [helper.py:55] xz-devel-5.2.4-3.el8.ppc64le
2021-11-08 14:58:35 DEBUG   : [helper.py:49] Running rpm -q python3-setuptools
2021-11-08 14:58:35 DEBUG   : [helper.py:55] python3-setuptools-39.2.0-6.el8.noarch
2021-11-08 14:58:35 DEBUG   : [helper.py:49] Running rpm -q numactl
2021-11-08 14:58:35 DEBUG   : [helper.py:55] numactl-2.0.12-13.el8.ppc64le
2021-11-08 14:58:35 DEBUG   : [helper.py:49] Running rpm -q policycoreutils-python-utils
2021-11-08 14:58:35 DEBUG   : [helper.py:55] policycoreutils-python-utils-2.9-15.el8.noarch
2021-11-08 14:58:35 DEBUG   : [helper.py:49] Running rpm -q openssl-devel
2021-11-08 14:58:35 DEBUG   : [helper.py:55] openssl-devel-1.1.1k-4.el8.ppc64le
2021-11-08 14:58:35 DEBUG   : [avocado-setup.py:181] Check if bootstrap required
2021-11-08 14:58:35 DEBUG   : [helper.py:49] Running which avocado
2021-11-08 14:58:35 DEBUG   : [helper.py:55] /usr/local/bin/avocado
2021-11-08 14:58:35 DEBUG   : [avocado-setup.py:197] Test needs to be downloaded/updated
2021-11-08 14:58:35 INFO    : [avocado-setup.py:406] Cleaning the Environment
2021-11-08 14:58:35 DEBUG   : [helper.py:47] Uninstalling avocado-framework
2021-11-08 14:58:35 DEBUG   : [helper.py:49] Running pip3 uninstall avocado-framework -y --disable-pip-version-check
2021-11-08 14:58:35 DEBUG   : [helper.py:55] Uninstalling avocado-framework-92.0:
  Successfully uninstalled avocado-framework-92.0
2021-11-08 14:58:35 DEBUG   : [helper.py:47] Uninstalling avocado-framework-plugin-varianter-yaml-to-mux
2021-11-08 14:58:35 DEBUG   : [helper.py:49] Running pip3 uninstall avocado-framework-plugin-varianter-yaml-to-mux -y --disable-pip-version-check
2021-11-08 14:58:36 DEBUG   : [helper.py:55] Uninstalling avocado-framework-plugin-varianter-yaml-to-mux-92.0:
  Successfully uninstalled avocado-framework-plugin-varianter-yaml-to-mux-92.0
2021-11-08 14:58:36 DEBUG   : [helper.py:47] Uninstalling avocado-framework-plugin-result-html
2021-11-08 14:58:36 DEBUG   : [helper.py:49] Running pip3 uninstall avocado-framework-plugin-result-html -y --disable-pip-version-check
2021-11-08 14:58:36 DEBUG   : [helper.py:55] Uninstalling avocado-framework-plugin-result-html-92.0:
  Successfully uninstalled avocado-framework-plugin-result-html-92.0
2021-11-08 14:58:36 DEBUG   : [helper.py:47] Uninstalling avocado-framework-plugin-vt
2021-11-08 14:58:36 DEBUG   : [helper.py:49] Running pip3 uninstall avocado-framework-plugin-vt -y --disable-pip-version-check
2021-11-08 14:58:37 DEBUG   : [helper.py:55] Cannot uninstall requirement avocado-framework-plugin-vt, not installed
2021-11-08 14:58:37 INFO    : [avocado-setup.py:301] Bootstrapping Framework
2021-11-08 14:58:37 INFO    : [avocado-setup.py:230]    1. Creating Avocado Config
2021-11-08 14:58:37 INFO    : [avocado-setup.py:307]    2. Installing Avocado Framework
2021-11-08 14:58:37 DEBUG   : [helper.py:47] Installing python package avocado-framework using pip
2021-11-08 14:58:37 DEBUG   : [helper.py:49] Running pip3 install -U avocado-framework
2021-11-08 14:58:41 DEBUG   : [helper.py:55] WARNING: Running pip install with root privileges is generally not a good idea. Try `pip3 install --user` instead.
Collecting avocado-framework
  Using cached https://files.pythonhosted.org/packages/b3/56/5d633f30547ec3d2cffdfcb01532fcda8ef82f21a9438ceb69a9bd5006b7/avocado_framework-92.0-py3-none-any.whl
Collecting setuptools (from avocado-framework)
  Downloading https://files.pythonhosted.org/packages/70/e9/84e2865fddfaba4506bc5d293d2a535bf27e31b12ca16d31564f8ce28cdb/setuptools-58.5.3-py3-none-any.whl (946kB)
Installing collected packages: setuptools, avocado-framework
  Found existing installation: setuptools 58.3.0
    Uninstalling setuptools-58.3.0:
      Successfully uninstalled setuptools-58.3.0
Successfully installed avocado-framework-92.0 setuptools-58.5.3
2021-11-08 14:58:41 DEBUG   : [helper.py:47] Installing python package avocado-framework-plugin-varianter-yaml-to-mux using pip
2021-11-08 14:58:41 DEBUG   : [helper.py:49] Running pip3 install -U avocado-framework-plugin-varianter-yaml-to-mux
2021-11-08 14:58:43 DEBUG   : [helper.py:55] WARNING: Running pip install with root privileges is generally not a good idea. Try `pip3 install --user` instead.
Collecting avocado-framework-plugin-varianter-yaml-to-mux
  Using cached https://files.pythonhosted.org/packages/92/61/8b7fc5fe34003706b5d0df652cbca4c5bca39a00c1535d0c9a95a5329880/avocado_framework_plugin_varianter_yaml_to_mux-92.0-py3-none-any.whl
Requirement already up-to-date: avocado-framework==92.0 in /usr/local/lib/python3.6/site-packages (from avocado-framework-plugin-varianter-yaml-to-mux)
Requirement already up-to-date: PyYAML>=4.2b2 in /usr/local/lib64/python3.6/site-packages (from avocado-framework-plugin-varianter-yaml-to-mux)
Requirement already up-to-date: setuptools in /usr/local/lib/python3.6/site-packages (from avocado-framework==92.0->avocado-framework-plugin-varianter-yaml-to-mux)
Installing collected packages: avocado-framework-plugin-varianter-yaml-to-mux
Successfully installed avocado-framework-plugin-varianter-yaml-to-mux-92.0
2021-11-08 14:58:43 DEBUG   : [helper.py:47] Installing python package avocado-framework-plugin-result-html using pip
2021-11-08 14:58:43 DEBUG   : [helper.py:49] Running pip3 install -U avocado-framework-plugin-result-html
2021-11-08 14:58:45 DEBUG   : [helper.py:55] WARNING: Running pip install with root privileges is generally not a good idea. Try `pip3 install --user` instead.
Collecting avocado-framework-plugin-result-html
  Using cached https://files.pythonhosted.org/packages/3d/55/7f3d62ca26df1db5a12e2040ea790f0014cec058bee31de140ea82de48c2/avocado_framework_plugin_result_html-92.0-py3-none-any.whl
Requirement already up-to-date: avocado-framework==92.0 in /usr/local/lib/python3.6/site-packages (from avocado-framework-plugin-result-html)
Requirement already up-to-date: jinja2 in /usr/local/lib64/python3.6/site-packages (from avocado-framework-plugin-result-html)
Requirement already up-to-date: setuptools in /usr/local/lib/python3.6/site-packages (from avocado-framework==92.0->avocado-framework-plugin-result-html)
Requirement already up-to-date: MarkupSafe>=2.0 in /usr/local/lib/python3.6/site-packages/MarkupSafe-2.0.1-py3.6-linux-ppc64le.egg (from jinja2->avocado-framework-plugin-result-html)
Installing collected packages: avocado-framework-plugin-result-html
Successfully installed avocado-framework-plugin-result-html-92.0
2021-11-08 14:58:45 DEBUG   : [helper.py:47] Creating test repo dir /root/s/tests/tests
2021-11-08 14:58:45 DEBUG   : [helper.py:49] Running mkdir -p /root/s/tests/tests
2021-11-08 14:58:45 DEBUG   : [helper.py:55] 
2021-11-08 14:58:45 INFO    : [helper.py:45]    3. Cloning/Updating the repo: avocado-misc-tests with branch master under /root/s/tests/tests/avocado-misc-tests
2021-11-08 14:58:45 DEBUG   : [helper.py:49] Running git clone https://github.com/avocado-framework-tests/avocado-misc-tests.git /root/s/tests/tests/avocado-misc-tests && cd /root/s/tests/tests/avocado-misc-tests && b=master;git reset --hard && git checkout master && git remote update && (git branch|grep $b||(git checkout $b && git switch -c $b))
2021-11-08 14:58:47 DEBUG   : [helper.py:55] Cloning into '/root/s/tests/tests/avocado-misc-tests'...
HEAD is now at 6bb4421 Merge pull request #2099 from vaishnavibhat/parm_file
Already on 'master'
Your branch is up to date with 'origin/master'.
Fetching origin
* master
2021-11-08 14:58:47 DEBUG   : [helper.py:49] Running which avocado
2021-11-08 14:58:47 DEBUG   : [helper.py:55] /usr/local/bin/avocado
2021-11-08 14:58:47 DEBUG   : [helper.py:49] Running /usr/local/bin/avocado list avocado-misc-tests/cpu/ppc64_cpu_test.py 2> /dev/null
2021-11-08 14:58:48 DEBUG   : [helper.py:55] 
2021-11-08 14:58:48 DEBUG   : [helper.py:49] Running /usr/local/bin/avocado list avocado-misc-tests/ras/ras.py 2> /dev/null
2021-11-08 14:58:49 DEBUG   : [helper.py:55] 
2021-11-08 14:58:49 DEBUG   : [helper.py:49] Running /usr/local/bin/avocado list avocado-misc-tests/perf/perfmon.py 2> /dev/null
2021-11-08 14:58:49 DEBUG   : [helper.py:55] 
2021-11-08 14:58:49 INFO    : [avocado-setup.py:350] 
2021-11-08 14:58:49 INFO    : [avocado-setup.py:363] Running Host Tests Suite mine_mine
2021-11-08 14:58:49 INFO    : [avocado-setup.py:375] Running: /usr/local/bin/avocado run --test-runner runner  avocado-misc-tests/cpu/ppc64_cpu_test.py avocado-misc-tests/ras/ras.py avocado-misc-tests/perf/perfmon.py --force-job-id 89b08f88eb331a94e78f51a7abfb0b4e44ab9a75                  --job-results-dir /root/s/tests/results  
2021-11-08 15:03:40 INFO    : [avocado-setup.py:386] 
2021-11-08 15:03:40 DEBUG   : [helper.py:49] Running grep 89b08f88eb331a94e78f51a7abfb0b4e44ab9a75 /root/s/tests/results/*/id|grep job-
2021-11-08 15:03:40 DEBUG   : [helper.py:55] /root/s/tests/results/job-2021-11-08T14.58-89b08f8/id:89b08f88eb331a94e78f51a7abfb0b4e44ab9a75
2021-11-08 15:03:40 INFO    : [avocado-setup.py:763] Summary of test results can be found below:
TestSuite                                                                                 TestRun    Summary
 
host_mine_mine                                                                            Run        Successfully executed
/root/s/tests/results/job-2021-11-08T14.58-89b08f8/job.log
| PASS 15 || CANCEL 0 || ERRORS 0 || FAILURES 0 || SKIP 0 || WARN 0 || INTERRUPT 0 |
2021-11-08 15:03:40 INFO    : [avocado-setup.py:766] Removing temporary mux dir
[root@system tests]#
```

**Pull request merged**